gsd/hoomd.py

```diff
@@ -1092,3 +1095,77 @@ def open(name, mode='rb'):
                          schema_version=[1, 4])
 
     return HOOMDTrajectory(gsdfileobj)
+
+
+def read_log(name, scalar_only=False):
+    """Read log from a hoomd schema GSD file into a dict of time-series arrays.
+
+    Args:
+        name (str): File name to open.
+        scalar_only (bool): Set to `True` to include only scalar log values.
+
+    Returns:
+        `dict`
+
+    Caution:
+        `read_log` requires that a logged quantity has the same shape in all
+        frames. Use `open` and `Snapshot.log` to read files where the shape
+        changes from frame to frame.
+
+    To create a *pandas* ``DataFrame`` with the logged data:
+
+    .. ipython:: python
+
+        import pandas
+
+        df = pandas.DataFrame(gsd.hoomd.read_log('log-example.gsd',
+                                                  scalar_only=True))
+        df
+    """
+    if fl is None:
+        raise RuntimeError("file layer module is not available")
+    if gsd is None:
+        raise RuntimeError("gsd module is not available")
+
+    with fl.open(name=str(name),
+                 mode='rb',
+                 application='gsd.hoomd ' + gsd.__version__,
+                 schema='hoomd',
+                 schema_version=[1, 4]) as gsdfileobj:
+
+        logged_data_names = gsdfileobj.find_matching_chunk_names('log/')
+        # Always log timestep associated with each log entry
+        logged_data_names.insert(0, 'configuration/step')
+        if len(logged_data_names) == 1:
+            raise RuntimeError('No logged data in file: ' + name)
```

Comment:
I am debating whether to just return an empty dict or one with `{"configuration/step": [0, 1, 2, ...]}`. I feel the second rather than erroring. I think that is what I would expect as a user.

Reply:
As a user I would expect an error or a warning. I think I might get confused if I didn't log timestep but got a dict of timesteps

Reply:
Yeah I was commenting on Brandon's suggestion - return a dict of timestep only if there's no other logged quantities

Reply:
I'm in favor of a warning instead of a `RuntimeError`. As you say, some users might wonder why there is no data in the dictionary. Other users may want to call `read_log` in batch on many files and should not need try/catch for those files missing data.

Reply:
I think that makes senes. But if we're changing this to a warning, should we then return a dict of `configuration/step` and document this?

Reply:
Yes, that is a good solution.